module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -650,6 +652,32 @@ namespace app
         return sys::msgHandled();
     }
 
+    sys::MessagePointer ApplicationCommon::handleAppShutdownRefresh(sys::Message *msgl)
+    {
+        auto *msg = static_cast<AppShutdownRefreshMessage *>(msgl);
+        assert(msg);
+        if (windowsStack.isEmpty()) {
+            LOG_ERROR("Current window is not defined");
+            return sys::msgNotHandled();
+        }
+        if (windowsStack.isEmpty() || (getCurrentWindow()->getName() != msg->getWindowName())) {
+            LOG_DEBUG("Ignore shutdown refresh request for window %s we are on window %s",
+                      msg->getWindowName().c_str(),
+                      windowsStack.isEmpty() ? "none" : getCurrentWindow()->getName().c_str());
+            return sys::msgNotHandled();
+        }
+
+        // send drawing commands only when if application is in active and visible.
+        auto window = getCurrentWindow();
```

Comment:
this comment is misleading, there was no check for active and visible here. Generally - it's not the place where it's checked either. What's mentioned in the comment is done in ApplicationManager.